akka-persistence/src/main/scala/akka/persistence/Persistence.scala

```diff
@@ -122,23 +124,52 @@ trait PersistenceStash extends Stash with StashFactory {
     Persistence(context.system).defaultInternalStashOverflowStrategy
 }
 
+trait RuntimePluginConfig {
+  /**
+   * Additional configuration of the journal plugin servicing this persistent actor.
+   * When empty, the whole configuration of the journal plugin will be taken from the [[Config]] loaded into the
+   * [[ActorSystem]].
+   * When configured, the journal plugin configuration will be taken from this [[Config]] merged with the [[Config]]
+   * loaded into the [[ActorSystem]].
+   *
+   * @return an additional configuration used to configure the journal plugin.
+   */
+  def journalPluginConfig: Config = ConfigFactory.empty
```

Comment:
I think it would be better if these were not implemented in the trait. Leaving them abstract instead. Then it's more clear what the end user must define when mixing in this trait. (and less risk of binary incompatibility issues for us in the future)